www/js/lib/uiUtil.js

```diff
@@ -46,30 +46,48 @@ define(rqDef, function(settingsStore) {
      * @param {Function} callback An optional function to call with the URI
      */
     function feedNodeWithBlob(node, nodeAttribute, content, mimeType, callback) {
+        var insertBlob = function (blob) {
+            var url = URL.createObjectURL(blob);
+            if (callback) callback(url);
+            node.addEventListener('load', function () {
+                URL.revokeObjectURL(url);
+            });
+            node.setAttribute(nodeAttribute, url);
+        };
         // Decode WebP data if the browser does not support WebP and the mimeType is webp
         if (webpMachine && /image\/webp/i.test(mimeType)) {
             // DEV: Note that webpMachine is single threaded and will reject an image if it is busy
             // However, the loadImagesJQuery() function in app.js is sequential (it waits for a callback
             // before processing another image) so we do not need to queue WebP images here
-            webpMachine.decode(content).then(function (uri) {
-                // DEV: WebpMachine.decode() returns a data: URI
-                // We callback before the node is set so that we don't incur slow DOM rewrites before processing more images
-                if (callback) callback(uri);
-                node.setAttribute(nodeAttribute, uri);
+            var canvas = document.createElement("canvas");
+            webpMachine.decodeToCanvas(canvas, content).then(function () {
+                if (canvas.msToBlob) {
+                    insertBlob(canvas.msToBlob());
+                } else if (canvas.mozGetAsFile) {
+                    
+                    // ** DEV: This is mozGetAsFile method - but results in blank images **
+                    // var f = canvas.mozGetAsFile('f.png');
+                    // insertBlob(f);
+
+                    // ** DEV: This is Insert canvas method - note that if we set the height or width directly, the imgaes get blanked **
+                    if (callback) callback(); // Calling back as soon as possible speeds up extraction
+                    canvas.style.width = node.getAttribute('width') + 'px';
+                    canvas.style.height = node.getAttribute('height') + 'px';
+                    node.parentNode.replaceChild(canvas, node);
+
+                } else {
+                    canvas.toBlob(function (imgBlob) {
+                        insertBlob(imgBlob);
+                    });
+                }
             }).catch(function (err) {
-                console.error('There was an error decoding image in WebpMachine', err);
-                if (callback) callback();
```

Comment:
Should revert this to previous catch code, it was more meaningful, plus callback is missing.